ext/telemetry/lib.rs

```diff
@@ -322,8 +343,69 @@ mod hyper_client {
   }
 }
 
-static OTEL_PROCESSORS: OnceCell<(SpanProcessor, LogProcessor)> =
-  OnceCell::new();
+enum MetricProcessorMessage {
+  ResourceMetrics(ResourceMetrics),
+  Flush(tokio::sync::oneshot::Sender<()>),
+}
+
+struct MetricProcessor {
+  tx: tokio::sync::mpsc::Sender<MetricProcessorMessage>,
+}
+
+impl MetricProcessor {
+  fn new(exporter: MetricExporter) -> Self {
+    let (tx, mut rx) = tokio::sync::mpsc::channel(2048);
+    let future = async move {
+      while let Some(message) = rx.recv().await {
+        match message {
+          MetricProcessorMessage::ResourceMetrics(mut rm) => {
+            if let Err(err) = exporter.export(&mut rm).await {
+              otel_error!(
+                name: "MetricProcessor.Export.Error",
+                error = format!("{}", err)
+              );
+            }
+          }
+          MetricProcessorMessage::Flush(tx) => {
+            if let Err(()) = tx.send(()) {
+              otel_error!(
+                name: "MetricProcessor.Flush.SendResultError",
+                error = "()",
+              );
+            }
+          }
+        }
+      }
+    };
+
+    (*OTEL_SHARED_RUNTIME_SPAWN_TASK_TX)
+      .unbounded_send(Box::pin(future))
+      .expect("failed to send task to shared OpenTelemetry runtime");
+
+    Self { tx }
+  }
+
+  fn submit(&self, rm: ResourceMetrics) {
+    let _ = self
+      .tx
+      .try_send(MetricProcessorMessage::ResourceMetrics(rm));
+  }
+
+  fn force_flush(&self) -> Result<(), anyhow::Error> {
+    let (tx, rx) = tokio::sync::oneshot::channel();
+    self.tx.try_send(MetricProcessorMessage::Flush(tx))?;
+    deno_core::futures::executor::block_on(rx)?;
```

Comment:
Using `block_on` is usually trouble, but I don't see any other way here...

Reply:
the other two processors use block_on as well